src/main/java/uk/co/codezen/maven/redlinerpm/mojo/AbstractRpmMojo.java

```diff
@@ -108,6 +108,12 @@
     protected boolean performCheckingForExtraFiles = true;
 
     /**
+     * Whether to replace already existing artifacts
+     */
+    @Parameter
+    protected boolean shouldReplaceArtifact = false;
```

Comment:
I think the reason why I added the check was to prevent an exception, though the correct default behaviour should be observed and on reflection the approach is wrong.

Perhaps change the code entirely to remove the need for the parameter and make it perform the right behaviour by deleting the artifact if present and bump the plugin version to 2.0.